quisp/modules/QRSA/ConnectionManager/ConnectionManager.cc

```diff
@@ -342,17 +344,16 @@ void ConnectionManager::respondToRequest(ConnectionSetupRequest *req) {
         pkr->setActual_destAddr(path.at(path.size() - 1));
         send(pkr, "RouterPort$o");
       }
-
     } else {
       EV_DEBUG << "Im not swapper!" << path.at(i) << "\n";
       int num_measure = req->getNum_measure();
 
       RuleSet *ruleset;
       int owner = path.at(i);
       if (i == 0) {  // if this is initiator
-        ruleset = generateTomographyRuleSet(owner, path.at(path.size() - 1), num_measure, qnics.at(qnics.size() - 1).fst.type, qnics.at(qnics.size() - 1).fst.index, num_resource);
+        ruleset = generateTomographyRuleSet(owner, path.at(path.size() - 1), num_measure, qnics.at(0).snd.type, qnics.at(0).snd.index, num_resource);
       } else {  // if this is responder
-        ruleset = generateTomographyRuleSet(owner, path.at(0), num_measure, qnics.at(0).snd.type, qnics.at(0).snd.index, num_resource);
+        ruleset = generateTomographyRuleSet(owner, path.at(0), num_measure, qnics.at(qnics.size() - 1).fst.type, qnics.at(qnics.size() - 1).fst.index, num_resource);
```

Comment:
why these lines changed? I want you to add comments that describe what `qnics.at(0).snd` or `qnics.at(qnics.size() - 1).fst` mean

Reply:
OK, let me add the comment